### PR TITLE
Feature/responses

### PR DESCRIPTION
**Pull request by nikumar1206**

### CODE REVIEW

File: app.go
Comment (tiredkangaroo) [inline app.go:1]: i understand we want to give users more control over the OpenAPI Spec but are we intergrating the openapi spec too much with the app?

File: context.go [hunk 1]
```diff
@@ -49,20 +49,20 @@ func (ctx *Context) Set(key string, value any) {
...
-// GetHeader gets the value of a request header with key k.
+// GetRequestHeader gets the value of a request header with key k.
 // It returns an empty string if not found.
-func (ctx *Context) GetHeader(k string) string {
+func (ctx *Context) GetRequestHeader(k string) string {
 	return ctx.Request.Header.Get(k)
 }
 
-// GetHeader gets the value of a response header with key k.
+// GetRequestHeader gets the value of a response header with key k.
```
Comment (tiredkangaroo) [inline context.go]: godoc comment doesn't match function

File: examples/restaurant_app/pasta.go [hunk 2]
```diff
@@ -22,16 +22,21 @@ var cheeses = map[int]string{
...
 }
 
+type FooResponse struct {
+	error   string
+	message string
+}
+
 type PastaCheeseInput struct {
-	ID int `kind:"path" description:"id of cheese"`
+	Id int `kind:"path" description:"id of cheese"`
```
Comment (tiredkangaroo) [inline examples/restaurant_app/pasta.go:31]: btw the struct tag supports the name field

File: field.go [hunk 3]
```diff
@@ -151,17 +148,17 @@ func validate(input map[string]any, schemaType reflect.Type) (bool, error) {
...
-		if required == true {
+		if required {
 			return false, ExpectedButNotFound(k)
 		}
 	}
 	return true, nil
 }
 
-// getHeaderParam gets the value of the param from the header. It may return error
+// GetRequestHeaderParam gets the value of the param from the header. It may return error
```
Comment (tiredkangaroo) [inline field.go]: no no no no no no no
this isn't a function to be made exported.

File: field.go [hunk 4]
```diff
@@ -173,7 +170,7 @@ func getQueryParam(c *Context, param Parameter) (string, error) {
 	return handleParam(value, param)
 }
 
-// getHeaderParam gets the value of the param from the cookie header.
+// GetRequestHeaderParam gets the value of the param from the cookie header.
 // It may return an error if it not found AND required.
```
Comment (tiredkangaroo) [inline field.go:174]: commend doesn't match function.

File: field.go [hunk 5]
```diff
@@ -173,7 +170,7 @@ func getQueryParam(c *Context, param Parameter) (string, error) {
 	return handleParam(value, param)
 }
 
-// getHeaderParam gets the value of the param from the cookie header.
+// GetRequestHeaderParam gets the value of the param from the cookie header.
```
Comment (tiredkangaroo) [inline field.go]: comment doesn't match function

File: field.go [hunk 6]
```diff
@@ -436,95 +455,12 @@ func newDefinition(schema any) Schema {
...
 		}
 		newDef.Properties[fieldName] = &nd
 	}
-	AddDefinition(st.Name(), newDef)
+	Definitions[st.Name()] = &newDef
 	newSchema.Ref = "#/definitions/" + st.Name()
 	return *newSchema
 }
-
-func handleInputSchema(pa *[]Parameter, rb *RequestBodyOrReference, s any) error { // should this return an error or should it panic?
```
Comment (tiredkangaroo) [inline field.go:449]: 😭😭

does this code still work? i feel like new bugs just happened cause of this removal.

File: middleware/logging.go [hunk 7]
```diff
@@ -26,13 +27,13 @@ var DefaultLoggingConfig LoggingConfig = LoggingConfig{
 		var statusColor = fmt.Sprintf(" %d ", sc)
 		switch {
 		case sc >= 500:
-			statusColor = color.ColorizeBold(string(sc), color.BgBrightRed, color.FgBlack)
+			statusColor = color.ColorizeBold(strconv.Itoa(sc), color.BgBrightRed, color.FgBlack)
```
Comment (tiredkangaroo) [inline middleware/logging.go]: could also just use fmt.Sprintf btw

File: openapi.go
Comment (tiredkangaroo) [inline openapi.go:1]: again are we implementing openapi.go into the entire puff thing too much?

File: response.go [hunk 8]
```diff
@@ -5,13 +5,59 @@ import (
...
+		ResponseType: responseType,
+	}
+}
+
 // Response is an interface that all response types should implement.
 type Response interface {
 	GetStatusCode() int
 	GetContentType() string
 	WriteContent(*Context) error
+	GetContent() any
```
Comment (tiredkangaroo) [inline response.go]: why does this exist? its quite unnecessary.

File: response.go [hunk 9]
```diff
@@ -76,6 +130,10 @@ func (f FileResponse) GetContentType() string {
 	return resolveContentType(f.ContentType, contentTypeFromFileName(f.FilePath))
 }
 
+func (f FileResponse) GetContent() any {
+	return f.FileContent
```
Comment (tiredkangaroo) [inline response.go]: again FileContent should not exist. WriteContent is meant to read the file at runtime.

File: response.go [hunk 10]
```diff
@@ -195,9 +257,13 @@ func (r RedirectResponse) GetContentType() string {
 	return "text/html; charset=utf-8"
 }
 
+func (r RedirectResponse) GetContent() any {
```
Comment (tiredkangaroo) [inline response.go]: if you have to return meaningless values for a function, just to implement an interface, than the interfacing is being too narrow. instead, functions that want to call GetContent() should just check the underlying type of the interface.

File: route.go [hunk 11]
```diff
@@ -28,3 +34,178 @@ func (r *Route) String() string {
...
+	route.fullPath = strings.Join(parts, "")
+}
+
+func (route *Route) createRegexMatch() {
+	escapedPath := strings.ReplaceAll(route.fullPath, "/", "\\/")
+	regexPattern := regexp.MustCompile(`\{[^}]+\}`).ReplaceAllString(escapedPath, "([^/]+)")
+	route.regexp = regexp.MustCompile("^" + regexPattern + "$")
+}
+
+func (route *Route) handleInputSchema() error { // should this return an error or should it panic?
```
Comment (tiredkangaroo) [inline route.go:54]: oh my golly gosh why couldn't we keep that in field.go. field.go is meant to handle input schemas. i see the overlapping purpose but field.go is a master at this exact thing.

File: app.go
Comment (nikumar1206) [reply app.go:1]: no the whole selling point is the auto schema generation
this isnt meant to be the fastest framework. 

just solid boring swagger pages

**Review by tiredkangaroo — APPROVED** [review]

**Pull request merged**